Atomics broken under Clang

Tip of master when brought over to OMPI:

/home/ubuntu/workspace/open-mpi.build.compilers/Compiler/clang38/src/3rd-party/openpmix/src/atomics/sys/atomic_stdc.h:250:5: error: member reference base type 'volatile void' is not a structure or union
    atomic_flag_clear ((volatile void *) lock);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/lib/llvm-3.8/bin/../lib/clang/3.8.0/include/stdatomic.h:181:63: note: expanded from macro 'atomic_flag_clear'
#define atomic_flag_clear(object) __c11_atomic_store(&(object)->_Value, 0, __ATOMIC_SEQ_CST)

--- a/src/atomics/sys/atomic_stdc.h
+++ b/src/atomics/sys/atomic_stdc.h
@@ -228,13 +228,21 @@ typedef atomic_flag pmix_atomic_lock_t;
 static inline void pmix_atomic_lock_init (pmix_atomic_lock_t *lock, bool value)
 {
     (void)value;
+#ifdef PMIX_HAVE_CLANG_BUILTIN_ATOMIC_C11_FUNC
+    atomic_flag_clear (lock);
+#else
     atomic_flag_clear ((volatile void *) lock);
+#endif
 }
 
 
 static inline int pmix_atomic_trylock (pmix_atomic_lock_t *lock)
 {
+#ifdef PMIX_HAVE_CLANG_BUILTIN_ATOMIC_C11_FUNC
+    return (int) atomic_flag_test_and_set (lock);
+#else
     return (int) atomic_flag_test_and_set ((volatile void *) lock);
+#endif
 }
 
 
@@ -247,7 +255,11 @@ static inline void pmix_atomic_lock(pmix_atomic_lock_t *lock)
 
 static inline void pmix_atomic_unlock (pmix_atomic_lock_t *lock)
 {
+#ifdef PMIX_HAVE_CLANG_BUILTIN_ATOMIC_C11_FUNC
+    atomic_flag_clear (lock);
+#else
     atomic_flag_clear ((volatile void *) lock);
+#endif
 }
 
 

